warehouse/slave/worker_test.go

```diff
@@ -714,264 +714,266 @@ func TestSlaveWorker(t *testing.T) {
-			existingDatatype: "json",
-			currentDataType:  "boolean",
-			value:            false,
-			newColumnVal:     "false",
+			name:              "should send json string values if existing datatype is json, new datatype is boolean",
+			existingDatatype:  "json",
+			currentDataType:   "boolean",
+			value:             false,
+			expectedColumnVal: "false",
 		},
 		{
-			name:             "should send json string values if existing datatype is jso, new datatype is int",
-			existingDatatype: "json",
-			currentDataType:  "int",
-			value:            1,
-			newColumnVal:     "1",
+			name:              "should send json string values if existing datatype is jso, new datatype is int",
+			existingDatatype:  "json",
+			currentDataType:   "int",
+			value:             1,
+			expectedColumnVal: "1",
 		},
 		{
-			name:             "should send json string values if existing datatype is json, new datatype is float",
-			existingDatatype: "json",
-			currentDataType:  "float",
-			value:            1.501,
-			newColumnVal:     "1.501",
+			name:              "should send json string values if existing datatype is json, new datatype is float",
+			existingDatatype:  "json",
+			currentDataType:   "float",
+			value:             1.501,
+			expectedColumnVal: "1.501",
 		},
 		{
-			name:             "should send json string values if existing datatype is json, new datatype is json",
-			existingDatatype: "json",
-			currentDataType:  "datetime",
-			value:            "2022-05-05T00:00:00.000Z",
-			newColumnVal:     `"2022-05-05T00:00:00.000Z"`,
+			name:              "should send json string values if existing datatype is json, new datatype is json",
+			existingDatatype:  "json",
+			currentDataType:   "datetime",
+			value:             "2022-05-05T00:00:00.000Z",
+			expectedColumnVal: `"2022-05-05T00:00:00.000Z"`,
 		},
 		{
-			name:             "should send json string values if existing datatype is json, new datatype is string",
-			existingDatatype: "json",
-			currentDataType:  "string",
-			value:            "string value",
-			newColumnVal:     `"string value"`,
+			name:              "should send json string values if existing datatype is json, new datatype is string",
+			existingDatatype:  "json",
+			currentDataType:   "string",
+			value:             "string value",
+			expectedColumnVal: `"string value"`,
 		},
 		{
-			name:             "should send json string values if existing datatype is json, new datatype is array",
-			existingDatatype: "json",
-			currentDataType:  "array",
-			value:            []any{false, 1, "string value"},
-			newColumnVal:     []any{false, 1, "string value"},
+			name:              "should send json string values if existing datatype is json, new datatype is string",
+			existingDatatype:  "json",
+			currentDataType:   "string",
+			value:             `{"json":"value"}`,
+			expectedColumnVal: `"{\"json\":\"value\"}"`,
```

Comment:
For the existing code, this will result in something like `"{"json":"value"}"` which will be invalid in case we are writing to a JSON output file (for example bigquery destination).